src/components/ImportRecordsFlow/stepThree/__test__/assignFieldModal.spec.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import AssignFieldModal from '../assignFieldModal';
+import getFileFieldsOptions from '../../helpers/getFileFieldsOptions';
+
+const attributes = {
+    name: { required: true },
+    email: {},
+};
+
+jest.mock('../../helpers/getFileFieldsOptions', () => jest.fn());
+
+describe('<AssignFieldModal />', () => {
+    it('should open modal when modal open status is set to true', () => {
+        const props = {
+            isAssignFieldModalOpen: true,
+            attributes,
+        };
+        const component = mount(<AssignFieldModal {...props} />);
+        expect(component.find('Modal').prop('isOpen')).toBe(true);
+    });
+    it('should run close method when cancel button is clicked', () => {
+        const onRequestCloseFn = jest.fn();
+        const props = {
+            isAssignFieldModalOpen: true,
+            attributes,
+            onRequestClose: onRequestCloseFn,
+        };
+        const component = mount(<AssignFieldModal {...props} />);
+        component
+            .find('AssignFieldModalFooter')
+            .find('button')
+            .at(0)
+            .simulate('click');
+        expect(onRequestCloseFn).toHaveBeenCalled();
+    });
+    it('should prepare file field options using columns param', () => {
+        const props = {
+            isAssignFieldModalOpen: true,
+            attributes,
+            columns: ['Name', 'Email'],
+        };
+        mount(<AssignFieldModal {...props} />);
+        expect(getFileFieldsOptions).toHaveBeenCalledWith(props.columns);
+    });
+    it('should have Preview component with their params properly passed', () => {
+        const props = {
+            isAssignFieldModalOpen: true,
+            databaseFieldToAssign: 'name',
+            data: [{ name: 'John', email: 'john@gmail.com' }],
+            attributes,
+        };
+        const component = mount(<AssignFieldModal {...props} />);
+        const preview = component.find('Preview');
+        const previewProps = {
+            field: preview.prop('field'),
+            data: preview.prop('data'),
+            attributes: preview.prop('attributes'),
+        };
+        const expectedProps = {
+            field: props.databaseFieldToAssign,
+            data: props.data,
+            attributes: props.attributes,
+        };
+        expect(previewProps).toEqual(expectedProps);
+    });
+});
```

Comment:
We should add test cases for:
- should pass the right options to Select component (mocking getFileFieldsOptions and returning something should match with select options)
- should set the right values in SelectedFieldsToAssign component when the param `fieldsMap[databaseFieldToAssign]` exists
- should set values in SelectedFieldsToAssign to a empty arrray when the param `fieldsMap[databaseFieldToAssign]` does not exists
- should fire onRequestClose when clicking assing button
- should fire onAssignField with the right values when clicking assing button

also try to simulate a selection in the Select component to test this:
```
    const selectFileField = event => {
        const { value } = event.target;
        setFileFieldsOptions(fileFieldsOptions.filter(field => field.value !== value));
        setFileFieldsToAssign([...fileFieldsToAssign, value]);
    };
```
by checking the new fileFieldsOptions passed to Select and the new values passed to SelectedFieldsToAssign

and the same above to test:
```
    const deleteFileField = value => {
        setFileFieldsOptions([
            ...fileFieldsOptions,
            {
                label: value,
                value,
            },
        ]);
        setFileFieldsToAssign(fileFieldsToAssign.filter(field => field !== value));
    };
```